### ISSUE TITLE
_and filter doesn't work as expected

### ISSUE THREAD

**Issue by fatihsolhan** [directus/directus#21169]
### Describe the Bug

Hi, I'll try my best to explain it with screenshots.

As you can see on the first query, even though I wrapped entire filter with `_and`, it still returns a response when it shouldn't since there is a `_neq` filter for date **2023-12-122**
![image](https://github.com/directus/directus/assets/22654053/dd54a979-8a8a-4d03-8011-fa92942dbe37)

Here is another approach for this query but still returns a data
![image](https://github.com/directus/directus/assets/22654053/c0af1f22-d37e-416a-9226-73186d503d37)

Lastly, I've tried to narrow the `_and` filter but I got an error this time
![image](https://github.com/directus/directus/assets/22654053/3a045708-ba06-4eac-b5f8-f2e5f0cae23b)

### To Reproduce

Just make a search for a date field with `_and` filter

### Directus Version

v10.8.2

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by br41nslug** [comment]
Instead of `_and`/`_neq` (which needs to be at the root level, https://docs.directus.io/reference/filter-rules.html#logical-operators) i think you're looking for `_none` or `_some` to filter relational data https://docs.directus.io/reference/filter-rules.html#some-vs-none-in-one-to-many

**Comment by fatihsolhan** [comment]
> Instead of `_and`/`_neq` (which needs to be at the root level, https://docs.directus.io/reference/filter-rules.html#logical-operators) i think you're looking for `_none` or `_some` to filter relational data https://docs.directus.io/reference/filter-rules.html#some-vs-none-in-one-to-many

Thanks for quick reply! I've put the `_and` filter to the root in the first screnshot, is it not correct?
Also, I saw `_none` and `_some` filters but unfortunately they don't exist for M2O relation.
![Screenshot 2024-01-23 at 17 06 24](https://github.com/directus/directus/assets/22654053/0ea1451d-9f43-4a46-a311-8170495d2ad4)

**Comment by br41nslug** [comment]
Judging from the returned results you're working with an many-to-many not a one-to-many right? (else there would be nothing to filter because an m2o can only have 1 item related to it)

**Comment by fatihsolhan** [comment]
Yeah, I am trying to filter inside the realtion. Since `deep` is natively supported with graphql queries, I don't think I need it, right? The structure is like this:

**Rentals**
- calendar: M2O to Calendars 👇🏻

**Calendars**
- calendarDay: M2M to CalendarDays 👇🏻

**CalendarDays**
- id
- date

**Comment by br41nslug** [comment]
You're very right indeed but graphql does requires it's own nested filters for that https://docs.directus.io/reference/query.html#examples-6

notice the filter on the relation not on the root item
![image](https://github.com/directus/directus/assets/9389634/52be56b0-60b3-4944-8cad-4215f54477ac)

**Comment by fatihsolhan** [comment]
hey @br41nslug 
I've simplified my collections and attached `calendarDays` to the `rentals` with M2M.

Here is my latest query and it doesn't work. I think it's not about `_and` filter anymore. I should be able to filter the `rentals` based on `calendarDays` value, that's why I've added filter to the `rentals`, not to the `calendarDays`

![Screenshot 2024-01-26 at 11 32 51](https://github.com/directus/directus/assets/22654053/a0ed9278-420a-4e73-baaf-f03a4529f0fe)

**Comment by br41nslug** [comment]
The result looks correct (even tho counter intuitive at first). This is where the previously mentioned `_some`/`_none` come in. `_some` is the default behavior for relational filters so your filter returns "calendar days that have _some calendarDays with a data _not_in "2024-01-01". Judging from your query you want this to behave like `_none` + `_nin`.

Not sure if you can currently achieve this with graphql https://github.com/directus/directus/issues/14228

**Comment by fatihsolhan** [comment]
Thank you for your help @br41nslug. Knowing that `_some` is the default behavior for relational filters made things more clear to me. I looked to the other issues and looks like the team working on a new tool called Durus which is I'm assuming (and hoping 🙏 ) gonna fix these issues.

I've tried another approach for this but it still didn't work as expected
![Screenshot 2024-01-27 at 19 51 28](https://github.com/directus/directus/assets/22654053/5476e644-862d-499d-b546-c23b1ee990e9)


Directus is a great tool and helped to me on lots of projects, I can't thank you guys enough for this awesome tool, but I have to say I'm very surprised that this relational filter is an issue with directus even though it's out there for years. I've planned everything relying on Directus because of it's awesome features (insights, flows, a great dashboard) and never thought I would face this type of issue

**Comment by sookoll** [comment]
> Directus is a great tool and helped to me on lots of projects, I can't thank you guys enough for this awesome tool, but I have to say I'm very surprised that this relational filter is an issue with directus even though it's out there for years. I've planned everything relying on Directus because of it's awesome features (insights, flows, a great dashboard) and never thought I would face this type of issue

I also struggle with simple search/filter on relational data. And it is so frustrating. I can't see where directus would fit at all. For very simple projects, this is overkill. And a bit bigger projects, directus is pain in the ***, You can't do things in a sane way.

**Comment by Nitwel** [comment]
May I ask what the underlying issue here is as I think it got lost in the discussions afterwards. To me it looks like that the issue could possibly be that we currently don't allow for `_some` and `_none` in GQL which will be fixed in https://github.com/directus/directus/pull/24644. Let me know if I'm on the right track here. 

With everything else, I think @br41nslug already explained quite well the behavior of o2m relations.

**Comment by br41nslug** [comment]
Correct Nils, I think the underlying issue is GraphQL not supporting the relational `_some/_none` filters limiting their flexibility with filters.
